blocks/library/button/index.js

```diff
@@ -98,6 +98,7 @@ class ButtonBlock extends Component {
 							color: textColor,
 						} }
 						keepPlaceholderOnFocus
+						inlineBlocksEnabled={ false }
```

Comment:
Should we let blocks disable this? I think it's kind of cool that you can insert inline images in any `RichText` field. I can even imagine this having use in captions, e.g. if I want to insert a little custom GIF emoji. I think we should give the users the freedom to decide, not the block, as it is still valid. Cc @jasmussen.

Reply:
I tend to agree that this is cool. We may find that we do want to allow a block to disable this, but it seems worth starting out liberal and then reining in once a need arises.

This partially depends on how well the combined inline and block level library works.

Reply:
Yeah I think it's worth to be liberal here and see what happens. Also `inlineBlocksEnabled` wouldn't work that well if we later add some other blocks that we then want here too.

Reply:
Ok I'm convinced 😃. I removed the `inlineBlocksEnabled` prop in https://github.com/WordPress/gutenberg/pull/5794/commits/51dc693ce8e6574da270e2e7e1bc4ce020e17ce5 so that users can choose to insert inline images in any RichText.

Currently the Image and Gallery blocks prevent the inline image from being resized, and set it to `display: block`, but I think I can fix these issues by adding CSS for the inline image.

The Text Column block has an issue where the inline image is duplicated into both columns. It looks like it is related to how `isSelected` is used, and similar to issue https://github.com/WordPress/gutenberg/issues/5633. Luckily, looks like this will be fixed in https://github.com/WordPress/gutenberg/pull/6280.

Reply:
Yes, should be fixed now. I just merged #6280 into `master` :)

Reply:
Awesome, I rebased and inline images are working correctly in the Text Column block now. Thanks @gziolo!

Reply:
On the surface I agree with the idea to allow the inline images everywhere, though I would be on the lookout for places where we might want only text.

also cc @mtias who had an opinion

Reply:
Makes sense. Inline images do look a bit out of place in the Gallery block:
<img width="640" alt="screen shot 2018-04-26 at 11 51 17 pm" src="https://user-images.githubusercontent.com/10903492/39343813-ed085f5c-49ac-11e8-8c15-ecdc22bf565a.png">

But perhaps there are use cases such as the custom GIF emoji as @iseulde mentioned where the user may want this option. And I can see why it may be good for the user to decide whether to insert it, rather than us deciding for them. 

Also, once user mentions are an inline block option, they may be appropriate for captions as well. But if we do come across places where we want to disable inline blocks, I'm happy to add the RichText prop back in.